### ISSUE TITLE
WinAppSdk release notes call out using wil's co_await DispatcherQueue support, but it doesn't exist

### ISSUE THREAD

**Issue by asklar** [microsoft/WindowsAppSDK#1552]
See microsoft/wil#213 - filing here to track

The code was removed when syncing with the OS copy of wil since it conflicted with some WRL names.

@dunhor @jevansaks FYI

**Comment by asklar** [comment]
@oldnewthing is fixing this in wil in microsoft/wil#214

**Comment by asklar** [comment]
The winmd has no dependency on anything C++ specific, it's just WinRT type metadata.
C++ projects will have to reference the wil nuget, and consume the header that Raymond is adding the definitions to.

**Comment by oldnewthing** [comment]
The old WindowsAppSDK contained custom headers that injected a `resume_foreground` into the `winrt` namespace, which was a violation of the C++/WinRT naming convention. Meanwhile, `wil` had a replacement `resume_foreground` that addressed many of the defects in the `winrt::resume_foreground`. In theory we could have duplicated them here (in a presumably new third namespace?), but then you run into the problem of keeping the two versions in sync.

Basically, `resume_foreground` is not a feature of the DispatcherQueue itself, but rather a new feature built on top of the DispatcherQueue. That new feature is implemented by WIL. (C++/WinRT learned the hard way that building features on top of Windows Runtime objects is a slippery slope, and has realized that it's better to leave the higher-order features to other libraries. "Do one thing well, don't try to do everything.")

**Comment by dotMorten** [comment]
@oldnewthing Thanks for the detailed answer. Much appreciated. Although I don't like the developer experience this creates, it does make a lot of sense

**Comment by LegendaryBlair** [comment]
Close this stale issue.
